client/connection.go

```diff
@@ -260,7 +260,7 @@ func (c *connection) readToFile() {
 					log.Warningf("unable to write to output %s for %s. %v", out.name, c.world.Name, err)
 				}
 			}
-			c.disconnect <- true
+			c.Close()
```

Comment:
Drive-by. Was hanging waiting for the connection to close if the connection had already dropped previously; explicitly close rather than just have the one channel.